apis/python/scripts/setup_build_env.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+# Setup script for vcpkg build environment (Linux/macOS)
+# This script bootstraps vcpkg and sets up environment variables
+# Usage: source this script or eval "$(./setup_build_env.sh -x)"
+
+set -eu -o pipefail
+
+# Get the repository root (assuming this script is in apis/python/scripts/)
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PYTHON_DIR="$(cd "${SCRIPT_DIR}/.." && pwd)"
+REPO_ROOT="$(cd "${PYTHON_DIR}/../.." && pwd)"
+
+# Check if vcpkg is already set up
+if [ -n "${CMAKE_TOOLCHAIN_FILE:-}" ] && [ -f "${CMAKE_TOOLCHAIN_FILE}" ]; then
+    # CMAKE_TOOLCHAIN_FILE is already set and exists - use it
+    echo "Using existing CMAKE_TOOLCHAIN_FILE: ${CMAKE_TOOLCHAIN_FILE}"
+    VCPKG_ROOT="$(dirname "$(dirname "${CMAKE_TOOLCHAIN_FILE}")")"
+elif [ -n "${VCPKG_ROOT:-}" ] && [ -d "${VCPKG_ROOT}" ]; then
+    # VCPKG_ROOT is set and exists - use it
+    echo "Using existing VCPKG_ROOT: ${VCPKG_ROOT}"
+    CMAKE_TOOLCHAIN_FILE="${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
+else
+    # Try to find vcpkg in common locations
+    if [ -d "${REPO_ROOT}/vcpkg" ]; then
+        VCPKG_ROOT="${REPO_ROOT}/vcpkg"
+    elif [ -d "${PYTHON_DIR}/../vcpkg" ]; then
+        # For cibuildwheel: vcpkg is cloned one level up from apis/python
+        VCPKG_ROOT="${PYTHON_DIR}/../vcpkg"
+    elif [ -d "${HOME}/vcpkg" ]; then
+        VCPKG_ROOT="${HOME}/vcpkg"
+    else
+        echo "Error: vcpkg not found. Please set VCPKG_ROOT or CMAKE_TOOLCHAIN_FILE environment variable."
+        echo "Searched locations:"
+        echo "  - ${REPO_ROOT}/vcpkg"
+        echo "  - ${PYTHON_DIR}/../vcpkg"
+        echo "  - ${HOME}/vcpkg"
+        echo ""
+        echo "To clone vcpkg, run:"
+        echo "  git clone https://github.com/microsoft/vcpkg.git ${REPO_ROOT}/vcpkg"
+        exit 1
+    fi
```

Comment:
[What the Core does](https://github.com/TileDB-Inc/TileDB/blob/e96ba69177d945ab318a59e7bf569b1139c6cfcc/cmake/Options/TileDBToolchain.cmake) is check if `CMAKE_TOOLCHAIN_FILE` is set, otherwise check for `ENV{VCPKG_ROOT}`, otherwise clone vcpkg using FetchContent (with an option to skip it and have the system supply all dependencies).